object_manager.go

```diff
@@ -562,6 +562,68 @@ func (objMgr *ObjectManager) DeleteCNAMERecord(ref string) (string, error) {
 	return objMgr.connector.DeleteObject(ref)
 }
 
+// Creates TXT Record. Use TTL of 0 to inherit TTL from the Zone
+func (objMgr *ObjectManager) CreateTXTRecord(recordname string, text string, ttl int, dnsview string) (*RecordTXT, error) {
+
+	recordTXT := NewRecordTXT(RecordTXT{
+		View: dnsview,
+		Name: recordname,
+		Text: text,
+		TTL:  ttl,
+	})
+
+	ref, err := objMgr.connector.CreateObject(recordTXT)
+	recordTXT.Ref = ref
+	return recordTXT, err
+}
+
+func (objMgr *ObjectManager) GetTXTRecordByRef(ref string) (*RecordTXT, error) {
+	recordTXT := NewRecordTXT(RecordTXT{})
+	err := objMgr.connector.GetObject(recordTXT, ref, &recordTXT)
+	return recordTXT, err
+}
+
+func (objMgr *ObjectManager) GetTXTRecord(name string) (*RecordTXT, error) {
+	if name == "" {
+		return nil, fmt.Errorf("name can not be empty")
+	}
+	var res []*RecordTXT
+
+	recordTXT := NewRecordTXT(RecordTXT{Name: name})
+
+	err := objMgr.connector.GetObject(recordTXT, "", &res)
+
+	if err != nil || res == nil {
+		return nil, err
+	}
+
+	return res[0], nil
+}
+
+func (objMgr *ObjectManager) UpdateTXTRecord(recordname string, text string) (*RecordTXT, error) {
+	var res []RecordTXT
+
+	recordTXT := NewRecordTXT(RecordTXT{Name: recordname})
+
+	err := objMgr.connector.GetObject(recordTXT, "", &res)
+
+	res[0].Text = text
```

Comment:
its better to have len(res) == 0 check here before doing `res[0].Text`

Reply:
@chinmayb great point, please check https://github.com/infobloxopen/infoblox-go-client/pull/89/commits/1a033234161e60a3df814ffe1ed714154289d451